third_party/gfxreconstruct/framework/decode/dive_block_data.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2025 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "decode/dive_block_data.h"
+#include "util/logging.h"
+
+#include <fstream>
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(decode)
+
+bool DiveBlockData::AddOriginalBlock(uint32_t index, uint64_t offset)
```

Comment:
maybe better to change `index` to type `size_t` since it is used to compare against vecter::size()